### PR TITLE
Fixed epp pod starting but not working when using multiple schedulingProfiles

### PR DESCRIPTION
**Pull request by learner0810**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fixed epp pod starting but not working when using multiple schedulingProfiles

**Which issue(s) this PR fixes**: 

Fixes # https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/1697

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by learner0810** [comment]
ping @ahg-g

**Comment by nirrozenbaum** [comment]
@learner0810 not sure if we want to test that on startup. we had that discussion in #1169, where I was actually suggesting to check it but was convinced by others that we shouldn’t.

there might be plugins combinations that work only if they are set together (e.g., in llm-d, pd profile handler will work correctly only if p filter is defined in prefill scheduling profile and d filter in decode profile). I think we don’t want to validate these kind of things cause it has no end. 
it should probably be the in the scope of a sanity test to catch those issues.

**Comment by learner0810** [comment]
> @learner0810 not sure if we want to test that on startup. we had that discussion in #1169, where I was actually suggesting to check it but was convinced by others that we shouldn’t.
> 
> there might be plugins combinations that work only if they are set together (e.g., in llm-d, pd profile handler will work correctly only if p filter is defined in prefill scheduling profile and d filter in decode profile). I think we don’t want to validate these kind of things cause it has no end. it should probably be the in the scope of a sanity test to catch those issues.

Even if misconfigured plugins can successfully route requests (even if routed to the wrong Pod), this may be tolerable. However, if misconfiguration causes requests to fail, we must test this during startup. 

Here is my understanding . Please correct me if I'm wrong.

**Comment by kfswain** [comment]
I think I'm okay with flagging this edge case. Esp because it references the default handler. Making our framework easier to use is a big plus wrt adoption.

/ok-to-test

**Comment by kfswain** [comment]
> I think we don’t want to validate these kind of things cause it has no end.

This is a fair point.... but the profile handler is kind of nuanced and a more complex part of our system. It is probably a real gotcha to try and do something sophisticated (multiple profiles) and then cut yourself on the no handler implementation.

**Comment by nirrozenbaum** [comment]
> > I think we don’t want to validate these kind of things cause it has no end.
> 
> This is a fair point.... but the profile handler is kind of nuanced and a more complex part of our system. It is probably a real gotcha to try and do something sophisticated (multiple profiles) and then cut yourself on the no handler implementation.

I’m fine with that. 
we just need to cut the validation somewhere. 
specifically if one configures handler incorrectly the logs show a very clear message about that. 
so from one hand it’s easy the identify, from the other hand it’s correct that we can fail on bootstrapping. 

for me it feels like testing the internals of plugins, but if you feel we want it on startup let’s go for it.

**Comment by kfswain** [comment]
/approve

Can we add a simple unit test to validate these changes? Thanks!

**Comment by learner0810** [comment]
> /approve
> 
> Can we add a simple unit test to validate these changes? Thanks!

Unit tests have been added. Please take another look.

File: pkg/epp/config/loader/configloader_test.go [hunk 1]
```diff
@@ -890,3 +895,25 @@ schedulingProfiles:
...
+plugins:
+- name: profileHandler
+  type: single-profile-handler
+- name: maxScore
+  type: max-score-picker
+schedulingProfiles:
+- name: default
+  plugins:
+  - pluginRef: maxScore
+  - pluginRef: profileHandler
```
Comment (nirrozenbaum) [inline pkg/epp/config/loader/configloader_test.go]: profile handler is not within the scope of schedulingProfile.. please remove:
```suggestion
```

File: pkg/epp/config/loader/configloader_test.go [hunk 2]
```diff
@@ -890,3 +895,25 @@ schedulingProfiles:
...
+- name: prof2
+  plugins:
+  - pluginRef: maxScore
+  - pluginRef: profileHandler
```
Comment (nirrozenbaum) [inline pkg/epp/config/loader/configloader_test.go]: ditto:
```suggestion
```

File: pkg/epp/config/loader/configloader_test.go (hunk 1 above)
Comment (learner0810) [reply pkg/epp/config/loader/configloader_test.go]: Thanks for the review. The changes have been committed.

**Comment by nirrozenbaum** [comment]
/lgtm
/approve

Thanks!
